client/src/controllers/TagController.ts

```diff
@@ -20,17 +21,31 @@ export class TagController extends Controller {
   static values = {
     options: { default: {}, type: Object },
     url: String,
+    debounce: { type: Number, default: 400 },
   };
 
   declare optionsValue: any;
   declare urlValue: any;
+  declare debounceValue: number;
   tagit?: JQuery<HTMLElement>;
 
   connect() {
     const preprocessTag = this.cleanTag.bind(this);
+    // Debounces by specified debounceValue to reduce unnecessary calls to database while user types, default is 300
+    const debouncedAutocomplete = debounce(
+      (request: { term: string }, response: (data: any) => void) => {
+        $.ajax({
```

Comment:
While this file does use a jQuery component, I think we should be careful to be adding more jQuery usage, could you please look at using `fetch` instead.

I know the syntax is a bit more verbose but it means that we can more easily migrate this to a non-jQuery solution in the future.

Reply:
I initially started out with fetch but due to the overwhelming usage of jQuery everywhere in the code base I didn't want to change things unnecessarily. I'll switch to using fetch.